nncf/experimental/onnx/graph/model_transformer.py

```diff
@@ -148,93 +149,117 @@ def select_model_inputs_outputs(model, outputs=None):
 
     def _apply_quantizer_insertion_transformations(self):
         # TODO: optimize
+        self.added_target_edges = Counter()
         for transformation in self.quantizer_insertion_commands:
             self._insert_quantizer_dequantizer(transformation)
 
-    def _insert_quantizer_dequantizer(self, transformation: ONNXQuantizerInsertionCommand):
-        # TODO (kshpv): remove many branches
-        # pylint: disable=too-many-branches
-        onnx_graph = ONNXGraph(self.transformed_model)
-        target_edge_names = set()
+    def _get_target_edge_name(self, transformation: ONNXQuantizerInsertionCommand, onnx_graph: ONNXGraph) -> Optional[
+        str]:
+        target_edge_name = None
         if transformation.target_point.type == TargetType.OPERATION_WITH_WEIGHTS:
             try:
-                target_edge_names.add(onnx_graph.get_weight_tensor_with_initializer(
-                    transformation.target_point.target_node_name))
+                target_edge_name = onnx_graph.get_weight_tensor_with_initializer(
+                    transformation.target_point.target_node_name)
             except RuntimeError as er:
```

Comment:
I partially agree with you. 
But look at the other point - if I use this function too much times in many lines of codes. In that case every time I have to check whether return value is None or not. 
At the same time, I know that if I use this function and it returns None, it is a problem in the algorithm and it can't be continued.
Assuming all of these I decided to use instead RuntimeError and if it is possible than there is no such tensor it could be interpreted by try-except
What do you think?

Reply:
>  In that case every time I have to check whether return value is None or not. At the same time, I know that if I use this function and it returns None, it is a problem in the algorithm and it can't be continued.

You literally handle the exception in the next line here and continue the algorithm in some way. If this is a fatal error, why handle the exception with `try... except` at all?

Reply:
The handling was removed. It was added by mistake